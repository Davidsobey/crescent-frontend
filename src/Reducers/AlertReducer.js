import AlertConstants from '../Constants/AlertConstants';

function AlertReducer(state = {}, action) {
  switch (action.type) {
    case AlertConstants.SUCCESS:
      return {
        type: 'alert-success',
        message: action.message,
        open: true,
      };
    case AlertConstants.ERROR:
      return {
        type: 'alert-danger',
        message: action.message,
        open: true,
      };
    case AlertConstants.CLEAR:
      return {};
    default:
      return state;
  }
}

export default AlertReducer;
