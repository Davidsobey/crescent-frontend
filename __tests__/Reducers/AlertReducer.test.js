// Reducer to be tested
import AlertReducer from '../../src/Reducers/AlertReducer';
import AlertConstants from '../../src/Constants/AlertConstants';

describe('AlertReducer', () => {
  test('ALERT_SUCCESS', () => {
    const action = { 
      type: AlertConstants.ALERT_SUCCESS, 
      message: 'success',
    };

    expect(AlertReducer(undefined, action)).toMatchSnapshot();
  });
});
