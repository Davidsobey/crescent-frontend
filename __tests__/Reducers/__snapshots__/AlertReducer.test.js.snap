// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertReducer ALERT_SUCCESS 1`] = `Object {}`;
